### PR TITLE
limit output of buildtest report

### LINKED ISSUE #1128: [FEATURE]: limit output of buildtest report
### Please describe your feature request

The `buildtest report` command will show all test results and the output can be quite long so for this feature we would like to limit the output by limiting the number of rows that get printed. 

We can have an option `buildtest report --count=<INT>` where `--count` expects a positive integer that limits the query to X number of rows.  

- [x] Add arg parse argument
- [x] Add implementation
- [x] Add bash completion
- [x] Add regression test
- [x] Add documentation
- [x] Add help config section

### Post question in Slack

- [X] I agree that I posted my question in [slack](https://hpcbuildtest.slack.com/) before creating this issue

### Is there an existing issue

- [X] I confirm there is no existing [issue](https://github.com/buildtesters/buildtest/issues) for this issue

### PR DESCRIPTION
**Pull request by Xiangs18**
Hi @shahzebsiddiqui, please see if further changes are required. Thank you!

Screen Shot 2022-06-30 at 11 35 06 AM

### CODE REVIEW

**Comment by shahzebsiddiqui** [comment]
@Xiangs18 this looks good. Can you share the output i think this looks perfect.

I presume this won't work with terse mode `buildtest rt --terse --count=1`

**Comment by Xiangs18** [comment]
Hi @shahzebsiddiqui, this won't work with terse mode buildtest rt --terse --count=1
However, it works with pager mode buildtest rt --pager --count 1
If you wanna make it work with terse mode as well, I can include it.

**Comment by Xiangs18** [comment]
Hi @shahzebsiddiqui, now it works with terse mode as well. Thank you!

File: tests/cli/test_report.py [hunk 1]
```diff
@@ -232,3 +232,25 @@ class args:
...
+        report_file = None
+        pager = None
+        terse = None
+        no_header = None
+        report_subcommand = None
+        helpfilter = False
+        helpformat = False
+        count = 5
+
+    report_cmd(args)
```
Comment (shahzebsiddiqui) [inline tests/cli/test_report.py]: Can you add another test with terse mode so just create another invocation to `args` . Alternately we may want to rewrite this test to directly call `Report` class directly. Either way works

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]

File: tests/cli/test_report.py (hunk 1 above)
Comment (Xiangs18) [reply tests/cli/test_report.py]: Screen Shot 2022-06-30 at 1 24 12 PM

**Comment by Xiangs18** [comment]
Screen Shot 2022-06-30 at 1 51 20 PM

**Review by shahzebsiddiqui — APPROVED** [review]
LGTM

**Pull request merged**